modules/platform/video/nuxeo-platform-video/src/main/java/org/nuxeo/ecm/platform/video/service/RecomputeVideoConversionsAction.java

```diff
@@ -0,0 +1,181 @@
+import org.nuxeo.ecm.platform.video.Video;
+import org.nuxeo.ecm.platform.video.VideoDocument;
+import org.nuxeo.lib.stream.computation.Topology;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.stream.StreamProcessorTopology;
+import org.nuxeo.runtime.transaction.TransactionHelper;
+
+/**
+ * BAF Computation that fills video renditions for the blob property described by the given xpath.
+ *
+ * @since 11.5
+ */
+public class RecomputeVideoConversionsAction implements StreamProcessorTopology {
+
+    private static final Logger log = LogManager.getLogger(RecomputeVideoConversionsAction.class);
+
+    public static final String ACTION_NAME = "recomputeVideoConversion";
+
+    public static final String ACTION_FULL_NAME = "bulk/" + ACTION_NAME;
+
+    public static final String PARAM_XPATH = "xpath";
+
+    public static final String PARAM_CONVERSION_NAMES = "conversionNames";
+
+    @Override
+    public Topology getTopology(Map<String, String> options) {
+        return Topology.builder()
+                       .addComputation(RecomputeRenditionsComputation::new, //
+                               Arrays.asList(INPUT_1 + ":" + ACTION_FULL_NAME, OUTPUT_1 + ":" + STATUS_STREAM))
+                       .build();
+    }
+
+    public static class RecomputeRenditionsComputation extends AbstractBulkComputation {
+
+        protected String xpath;
+
+        protected List<String> conversionNames;
+
+        protected VideoService videoService = Framework.getService(VideoService.class);
+
+        public RecomputeRenditionsComputation() {
+            super(ACTION_FULL_NAME);
+        }
+
+        @Override
+        public void startBucket(String bucketKey) {
+            BulkCommand command = getCurrentCommand();
+            xpath = command.getParam(PARAM_XPATH);
+            conversionNames = command.getParam(PARAM_CONVERSION_NAMES);
+            if (conversionNames.isEmpty()) {
+                conversionNames = videoService.getAvailableVideoConversions()
+                                              .stream()
+                                              .map(VideoConversion::getName)
+                                              .collect(Collectors.toList());
+            }
+        }
+
+        @Override
+        protected void compute(CoreSession session, List<String> ids, Map<String, Serializable> properties) {
+            log.debug("Compute action: {} generating conversions: {} for doc ids: {}", ACTION_NAME, conversionNames,
+                    ids);
+            for (String docId : ids) {
+                if (!session.exists(new IdRef(docId))) {
+                    log.debug("Doc id doesn't exist: {}", docId);
+                    continue;
+                }
+
+                DocumentModel workingDocument = session.getDocument(new IdRef(docId));
+                Property fileProp = workingDocument.getProperty(xpath);
+                if (!((fileProp instanceof Blob) || (fileProp instanceof BlobProperty))) {
```

Comment:
Remove extra parentheses: `if (!(fileProp instanceof Blob || fileProp instanceof BlobProperty))`

But `fileProp instanceof Blob` is not possible, we always get a kind `Property` not a value.